From e7f16327176447ab153cbaf0a547f65364ac1d7d Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sun, 28 Oct 2018 17:10:00 -0500
Subject: [PATCH 2/2] Fix inverted size check

error case is if offset+size goes beyond
end of file, not if it's within file bounds.
---
 elf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf.c b/elf.c
index 12ca448..3e80fd8 100644
--- a/elf.c
+++ b/elf.c
@@ -36,7 +36,7 @@ static int process_phead(void *elf, size_t size, ElfN_Phdr *header)
 	case PT_INTERP:
 		if (mode == MODE_READ && header->p_memsz > 0) {
 			fputs(" INTERP=", stdout);
-			if(header->p_offset + header->p_memsz < size) {
+			if(header->p_offset + header->p_memsz >= size) {
 				fputs("INVALID", stdout);
 				return -1;
 			}
-- 
2.19.1

